module/surveySubmissions/helper.js

```diff
@@ -0,0 +1,421 @@
+                        "evidences." + evidenceId + ".submissions",
+                        "status",
+                        "createdBy"
+                    ]
+                );
+                
+                if (!submissionDocument.length) {
+                    throw new Error(messageConstants.apiResponses.SUBMISSION_NOT_FOUND)
+                }
+
+                if (submissionDocument[0].status == messageConstants.common.SUBMISSION_STATUS_COMPLETED &&
+                    submissionDocument[0].createdBy == userId) {
+                    throw new Error(messageConstants.apiResponses.MULTIPLE_SUBMISSIONS_NOT_ALLOWED)
+                }
+               
+                if (submissionDocument[0].evidences[evidenceId].isSubmitted && submissionDocument[0].evidences[evidenceId].isSubmitted == true) {
+                    submissionDocument[0].evidences[evidenceId].submissions.forEach(submission => {
+                        if (submission.submittedBy == userId) {
+                            result.allowed = false;
+                        }
+                    })
+                }
+
+                return resolve({
+                    success: true,
+                    message: messageConstants.apiResponses.SURVEY_SUBMISSION_CHECK,
+                    data: result
+                });
+            }
+            catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                })
+            }
+        })
+    }
+
+
+     /**
+    * List created and submitted surveys.
+    * @method
+    * @name list
+    * @param {String} userId - logged in userId
+    * @returns {Json} - survey list.
+    */
+
+    static list(userId = "") {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if (userId == "") {
+                    throw new Error(messageConstants.apiResponses.USER_ID_REQUIRED_CHECK)
+                }
+
+                let getSurveyList = [
+                    solutionsHelper.solutionDocumentsByAggregateQuery
+                    (
+                            [
+                                {
+                                    "$match": {
+                                        "author": userId,
+                                        "type": messageConstants.common.SURVEY,
+                                        "isDeleted": false,
+                                    }
+                                },
+                                {
+                                    "$project": {
+                                        "solutionId": "$_id",
+                                        "name": 1,
+                                        "status": 1
+
+                                    }
+                                },
+                                { "$sort": { createdAt: 1 } }
```

Comment:
@deepakr123 - Has to be descending order.

Reply:
resolved